### DISCUSSION TITLE
Por que fazer a api em javascript?

### DISCUSSION THREAD

**Discussion by adeirjunior** [BrasilAPI/BrasilAPI#629] [Q&A] [answered]
Sabem dizer o por que de terem decidido fazer a BrasilAPI em javascript? não seria mais seguro usar typescript por ser uma linguagem com tipagem e adicionar uma camada a mais de segurança ao projeto?

**Comment by RodriAndreotti** [comment]
Boa noite, @adeirjunior!
Tudo bem?

Não sou o idealizador do projeto, mas como uma pessoa que tenta ajudar sempre que possível acho que seria mais uma questão de barreira de entrada para desenvolvedores que estejam iniciando sua jornada.

E, se pararmos para analisar, typescript não deixa de ser javascript.

**Reply by adeirjunior** [reply]
Sim, entendo que acaba sendo mais fácil pra alguns. Mas não deixa de dar uma dificuldade maior na hora de corrigir bugs que possam ocorrer, ainda mais pra uma API onde é preciso ter certeza que os dados que estão chegando são realmente aqueles requisitados.

**Reply by RodriAndreotti** [reply, marked as answer]
Neste ponto eu acho que é mais questão pessoal do que tudo, eu pessoalmente não vejo dessa forma.
No meu ver o que o typescript faz é adicionar uma camada de abstração emulando uma tipagem forte ao javascript, basicamente tenta inferir a linguagem um comportamento que não é o padrão da linguagem.
Isso tem seus prós e seus contras.

É possível garantir que os tipos de dados que vc está recebendo ou enviando sejam exatamente o que se espera sem a necessidade do Typescript

**Reply by adeirjunior** [reply]
Bom, isso é verdade kkk, acho que é mais questão de costume mesmo.
